app/forms/thredded/edit_topic_form.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+module Thredded
+  class EditTopicForm
+    include ActiveModel::Model
+
+    delegate :title, :category_ids, :locked, :sticky, :messageboard, :messageboard_id, :valid?,
+             to: :@topic
+
+    # @param user [Thredded.user_class]
+    # @param topic [Thredded::Topic]
+    def initialize(user:, topic:)
+      @user = user
+      @topic = topic
+    end
+
+    def self.model_name
+      Thredded::Topic.model_name
+    end
+
+    def category_options
+      @topic.messageboard.categories.map { |cat| [cat.name, cat.id] }
+    end
+
+    def messageboard_options
+      @user.thredded_can_write_messageboards.map { |messageboard| [messageboard.name, messageboard.id] }
```

Comment:
FeatureEnvy: Thredded::EditTopicForm#messageboard_options refers to 'messageboard' more than self (maybe move it to another class?)